internal/auth/internal/types/errors.go

```diff
@@ -0,0 +1,11 @@
+package types
+
+import "errors"
+
+var (
+	errInvalidAction         = errors.New("action must be either allow or deny")
+	errEmptyDestinationRules = errors.New("permissions must contain at least one destination rule")
+	errSourcesTenancy        = errors.New("permissions sources may not specify partitions, peers, and sameness_groups together")
+	errInvalidPrefixValues   = errors.New("prefix values must be valid and not combined with explicit names")
+	errEmptySources          = errors.New("permissions must contain at least one source")
```

Comment:
You should also be able to create a traffic permission without a source (e.g. https://docs.google.com/document/d/1XWZoGwMYV0VDZlssiCRRkvea7r6tD2VZZoaDOYjBudc/edit#heading=h.uev1phnt8wck)